src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs

```diff
@@ -773,29 +763,127 @@ private void TasView_MouseDown(object sender, MouseEventArgs e)
 			}
 		}
 
+		/// <summary>
+		/// Begins a batch of edits, for auto-restore purposes. Auto-restore will be delayed until EndBatchEdit is called.
+		/// </summary>
+		private void BeginBatchEdit()
+		{
+			_batchEditing = true;
+		}
+		/// <returns>Returns true if the input list was redrawn.</returns>
+		private bool EndBatchEdit()
+		{
+			_batchEditing = false;
+			if (_batchEditMinFrame != -1)
+			{
+				return FrameEdited(_batchEditMinFrame);
+			}
+
+			return false;
+		}
+
+		public void BatchEdit(Action action)
+		{
+			// This is only caled from Lua.

```

Comment:
If it's only called from Lua, put "Lua" or "ApiHawk" in the method name.
(Per the other thread, I think it's fine to keep the separate begin/end calls everywhere else.)